protocol-units/bridge/integration-tests/tests/movement_eth.rs

```diff
@@ -0,0 +1,463 @@
+				.faucet_client()
+				.expect("Failed to get FaucetClient")
+				.write()
+				.unwrap();
+			faucet_client.fund(movement_client_signer.address(), 100_000_000_000).await?;
+
+			let balance =
+				coin_client.get_account_balance(&movement_client_signer.address()).await?;
+			assert!(
+				balance >= 100_000_000_000,
+				"Expected Movement Client to have at least 100_000_000, but found {}",
+				balance
+			);
+		} // End of immutable borrow scope
+
+		// Mutable borrow to initiate the bridge transfer
+		{
+			let movement_client =
+				harness.movement_client_mut().expect("Failed to get MovementClient");
+			let _ = movement_client.publish_for_test();
+
+                        let mint_amount = 200 * 100_000_000; // Assuming 8 decimals for MovETH
+
+                        let mint_args = vec![
+                                utils::serialize_address_initiator(&movement_client.signer().address())?, // Mint to initiator's address
+                                utils::serialize_u64_initiator(&mint_amount)?,                     // Amount to mint (200 MovETH)
+                        ];
+         
+                        let mint_payload = utils::make_aptos_payload(
+                                movement_client.counterparty_address, // Address where moveth module is published
+                                "moveth",
+                                "mint",
+                                Vec::new(),
+                                mint_args,
+                        );
+        
+                        utils::send_and_confirm_aptos_transaction(&movement_client.rest_client, movement_client.signer(), mint_payload)
+                                .await
+                                .map_err(|_| BridgeContractInitiatorError::MintError)?; // New error variant for mint failure
+        
+                        debug!("Successfully minted 200 MovETH to the initiator");
+
+			movement_client
+				.initiate_bridge_transfer(
+					InitiatorAddress(MovementAddress(initiator_addr)),
+					RecipientAddress(recipient.clone()),
+					HashLock(hash_lock),
+					TimeLock(time_lock),
+					Amount(AssetType::Moveth(amount)),
+				)
+				.await
+				.expect("Failed to initiate bridge transfer");
+		} // End of mutable borrow scope
+
+		// Immutable borrow to extract the bridge transfer ID from the transaction
+		let movement_client = harness.movement_client_mut().expect("Failed to get MovementClient");
+		let sender_address = movement_client.signer().address();
+		let sequence_number = 0; // Replace this with the correct sequence number
+
+		let mut bridge_transfer_id: Option<String> = None;
+
+		let rest_client = movement_client.rest_client();
+		let transactions = rest_client
+			.get_account_transactions(sender_address, Some(sequence_number), Some(20))
+			.await
+			.map_err(|e| anyhow::Error::msg(format!("Failed to get transactions: {:?}", e)))?;
+
+		if let Some(transaction) = transactions.into_inner().last() {
```

Comment:
Let's put this in a utils/ inside integration tests. I don't see it as "belonging" to `TestHarness`. Currently it obstructs the main logic of the code.